src/device-state/current-state.ts

```diff
@@ -2,55 +2,47 @@ import * as _ from 'lodash';
 import * as url from 'url';
 import { CoreOptions } from 'request';
 
-import * as constants from '../lib/constants';
+import { backoffPromise } from '../lib/backoff';
```

Comment:
Moved backoff to it's own module since it could be useful for other areas of the codebase (like adding it to target-state fetching since it uses its own mechanism)

Reply:
cc @toochevere